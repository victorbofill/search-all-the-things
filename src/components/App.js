import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom';
import './App.css';
import logo from './logo.jpg';
import Paging from './Paging';
import { searchCharacters } from '../services/rmAPI';
import SearchCharacters from './SearchCharacters';
import CharacterResults from './CharacterResults';

export default class App extends Component {
  state = {
    loading: false,
    character: '',
    error: null,
    page: 1,
    characterResults: [],
    totalResults: null,
    pages: null
  };

  searchCharacters = () => {
    const { character, page } = this.state;

    this.setState({ loading: true });

    searchCharacters({ character }, { page })
      .then(({ info, results }) => {
        this.setState({ characterResults: results, error: null, totalResults: info.count, pages: info.pages });
      }, error => {
        this.setState({ error });
      })
      .then(() => this.setState({ loading: false }));
  };

  handleSearchCharacters = ({ search }) => {
    this.setState({ character: search }, this.searchCharacters);
  };

  handlePage = ({ page }) => {
    this.setState({ page }, this.searchCharacters);
  };

  render() {
    const { loading, characterResults, error, totalResults, page, character } = this.state;

    return (
      <main>
        <header><img src={logo} /></header>
        <fieldset>
          <div className="search-characters-container">
            <SearchCharacters onSearch={this.handleSearchCharacters}/>
          </div>
        </fieldset>
        <section>
          {character && 
                      <Paging
                        totalResults={totalResults}
                        page={page}
                        search={character}
                        onPage={this.handlePage}
                      />
          }
        </section>
        <section className="notifications">
          {loading && <div className="loading">Loading...</div>}
          {error && <div>Error: {error.message}</div>}
        </section>
        <section className="character-results">
          <CharacterResults characterResults={characterResults}/>
        </section>
      </main>
    );
  }
}