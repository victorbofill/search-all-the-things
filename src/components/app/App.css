header {
  height: 10em;
  margin-bottom: 1em;
  max-width: 100%;
}

header img {
  max-height: 10em;
  display: block;
}

.loading {
  background-color: white;
  font-size: 2em;
  font-weight: bold;
  height: 100%;
  opacity: .75;
  position: absolute;
  width: 100%;
  z-index: 1;
}