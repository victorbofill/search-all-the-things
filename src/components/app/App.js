import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Header from './Header';
import Home from '../home/Home';
import Characters from '../characters/Characters';
import Episodes from '../episodes/Episodes';
import Locations from '../locations/Locations';

import './App.css';

// TODO:
// Create router to switch between search criteria, with nav bar
// Search functionality for all three criteria

export default class App extends Component {

  render() {
    return (
      <Router>
        <div>
          <Header />
          <main>
            <Switch>
              <section>
                <Route strict path="/" component={Home}/>
                <Route path="/characters" component={Characters}/>
                <Route path="/episodes" component={Episodes}/>
                <Route path="/locations" component={Locations}/>
                <Redirect to="/"/>
              </section>
            </Switch>
          </main>
        </div>
      </Router>
    );
  }
}