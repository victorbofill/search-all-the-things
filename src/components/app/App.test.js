import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';

describe('App', () => {
  test('with home', () => {
    const wrapper = shallow(<App home={true}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});