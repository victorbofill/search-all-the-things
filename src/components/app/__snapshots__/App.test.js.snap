// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App with home 1`] = `
<BrowserRouter>
  <div>
    <Header />
    <main>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          path="/characters/:id"
          render={[Function]}
        />
        <Route
          component={[Function]}
          path="/characters"
        />
        <Route
          component={[Function]}
          path="/episodes/:id"
        />
        <Route
          component={[Function]}
          path="/episodes"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/locations"
        />
        <Redirect
          push={false}
          to="/"
        />
      </Switch>
    </main>
  </div>
</BrowserRouter>
`;
